spec.html

```diff
@@ -19473,17 +19511,103 @@
+            </td>
+            <td>
+              a parse result
+            </td>
+            <td>
+              The result of parsing the source text of this module using |Script| as the goal symbol.
+            </td>
+          </tr>
+          <tr>
+            <td>
+              [[HostDefined]]
+            </td>
+            <td>
+              Any, default value is *undefined*.
+            </td>
+            <td>
+              Field reserved for use by host environments that need to associate additional information with a script.
+            </td>
+          </tr>
+          </tbody>
+        </table>
+      </emu-table>
+    </emu-clause>
+
+    <emu-clause id="sec-parse-script" aoid="ParseScript">
+      <h1>ParseScript ( _sourceText_, _hostDefined_ )</h1>
+
+      <p>The abstract operation ParseScript with arguments _sourceText_ and _hostDefined_ creates a Script Record based upon the result of parsing _sourceText_ as a |Script|. ParseScript performs the following steps:</p>
+
+      <emu-alg>
+        1. Assert: _sourceText_ is an ECMAScript source text (see clause <emu-xref href="#sec-ecmascript-language-source-code"></emu-xref>).
+        1. Parse _sourceText_ using |Script| as the goal symbol and analyze the parse result for any Early Error conditions. If the parse was successful and no early errors were found, let _body_ be the resulting parse tree. Otherwise, let _body_ be an indication of one or more parsing errors and/or early errors. Parsing and early error detection may be interweaved in an implementation dependent manner. If more than one parse or early error is present, the number and ordering of reported errors is implementation dependent but at least one error must be reported.
+        1. If _body_ is an abrupt completion or error indication, then
+          1. Throw a SyntaxError exception.
+        1. Return Script Record {[[Realm]]: *undefined*, [[Environment]]: *undefined*, [[ECMAScriptCode]]: _body_, [[HostDefined]]: _hostDefined_}.
+      </emu-alg>
+      <emu-note>
+        <p>An implementation may parse script source text and analyze it for Early Error conditions prior to evaluation of ParseScript for that script source text. However, the reporting of any errors must be deferred until the point where this specification actually performs ParseScript upon that source text.</p>
+      </emu-note>
+    </emu-clause>
+
     <!-- es6num="15.1.7" -->
-    <emu-clause id="sec-runtime-semantics-scriptevaluation">
-      <h1>Runtime Semantics: ScriptEvaluation</h1>
-      <p>With argument _realm_.</p>
-      <emu-grammar>Script : ScriptBody?</emu-grammar>
+    <emu-clause id="sec-runtime-semantics-scriptevaluation" aoid="ScriptEvaluation">
+      <h1>ScriptEvaluation ( _scriptRecord_ )</h1>
+
       <emu-alg>
-        1. If |ScriptBody| is not present, return NormalCompletion(~empty~).
-        1. Let _globalEnv_ be _realm_.[[globalEnv]].
+        1. Let _globalEnv_ be _scriptRecord_.[[Realm]].[[globalEnv]].
```

Comment:
Perfect, this was the kind of thing I was unsure on. I'll add new semantics for Script: [empty].